#6: Recent changes by N4JS committers

jpilgrim opened the pull request:
Changes added by N4JS committers after the initial contribution was submitted.

jpilgrim commented:
This commit contains files edited by Luca, we need him to "sign" the ECA. His participation will then be marked in the Squash-Commit message.

qtran-n4 commented on tests/org.eclipse.n4js.lang.tests/src/org/eclipse/n4js/tests/parser/ES_07_09_AutomaticSemicolonInsertionParserTest.xtend:
```diff
@@ -138,6 +138,50 @@ class ES_07_09_AutomaticSemicolonInsertionParserTest extends AbstractParserTest
+	 * will insert semicolons after the document.write() call and after the .forEach() call, but not before the array
+	 * literal. In fact, it will attempt to use the array literal as indexer operator to the string from the line before.
+	 *
+	 * (c) Copyright 2014 by Armin Ronacher.
```
This class contains third-party code. Should we check it in?

yoosiba replied:
This is actually required by the headless compiler, other wise it will go off rails when running.
We can not commit it, but then we cannot build our our headless tool.

jpilgrim replied:
Sent an email to the author. I hope that he will help us (signing off a PR). Otherwise we have to remove that code or replace it with other tests.

qtran-n4 commented on tools/org.eclipse.n4js.hlc/src/main/resources/org/eclipse/emf/ecore/plugin/plugin.properties:
```diff
@@ -0,0 +1,220 @@
+
+
+# /**
+#  * Copyright (c) 2002-2010 IBM Corporation and others.
```
This file's copyright is IBM, have you decided how to deal with this plugin.properties file?

yoosiba replied:
This is actually required by the headless compiler, other wise it will go off rails when running.
We can not commit it, but then we cannot build our our headless tool.

jpilgrim replied:
see below, should be ok

jpilgrim commented:
@qtran-n4 , @yoosiba : About the emf properties file: This files stems from the EMF project, doesn't it? IMHO it is ok IP wise to add it without a CQ. But we should mark it accordingly.

jpilgrim commented:
@qtran-n4 I mailed the author of the blog posting asking whether he could sign the ECA. Alas he refused my request, stating that there is no copyright needed for these snippets. Thus I removed the examples taken from his blog post and replaced them with similar (and more simpler) versions. I left the link to his blog post in the test though.

jpilgrim commented:
@qtran-n4 Your last PR (456) is not contained in this patch since you pushed it _after_ we created the diff. There will probably more changes before we switch to eclipse/n4js, thus we will need another patch PR later on anyway.

yoosiba commented:
We just got access to the eclipse jenkins instance for n4js and it is quite different from what we used so far. It will take some time to setup build pipeline in that infrastructure. Also Jenkinsfile itself will need to be modified a bit. I would propose to skip it in current PRs, and work on it separately.

i.e. please blacklist Jenkinsfile in the script that prepares PR. New Jenkins file will be prepared directly on eclipse/n4js codebase
